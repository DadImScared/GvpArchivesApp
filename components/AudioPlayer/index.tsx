import * as React from "react";
import { StyleSheet, View } from "react-native";

import { withAudioPlayer } from "../withAudioPlayer";
import ButtonGroup from "./ButtonGroup";
import DurationSeeker from "./DurationSeeker";
import { IAudioPlayerState } from "./Provider";
import Title from "./Title";

interface IProps {
  audio: IAudioPlayerState;
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: "white",
    bottom: 0,
    padding: 8,
    position: "absolute",
    width: "100%",
  }
});

export class AudioPlayer extends React.Component<IProps> {
  render() {
    const {
      audio: {
        isSeeking,
        isPlaying,
        onPlayPausePressed,
        getSongName,
        getSeekSliderPosition,
        getTimeStamp,
        onSeekSliderChange,
        onSeekSliderComplete

      }
    } = this.props;
    return (
      <View style={styles.wrapper}>
        <Title getSongName={getSongName} />
        <DurationSeeker
          isSeeking={isSeeking}
          getSeekSliderPosition={getSeekSliderPosition}
          getTimeStamp={getTimeStamp}
          onSeekSliderChange={onSeekSliderChange}
          onSeekSliderComplete={onSeekSliderComplete}
        />
        <ButtonGroup
          onPlayPausePressed={onPlayPausePressed}
          isPlaying={isPlaying}
        />
      </View>
    );
  }
}

export default withAudioPlayer(AudioPlayer) as any; // property error without "any"
