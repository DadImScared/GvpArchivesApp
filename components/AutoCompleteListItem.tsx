import * as React from "react";
import { Text } from "react-native";

import { ListItem } from "native-base";

interface IProps {
  text: string;
}

export class AutoCompleteListItem extends React.Component<IProps> {

  render() {
    return (
      <ListItem button={true} noIndent={true}>
        <Text>{this.props.text}</Text>
      </ListItem>
    );
  }
}

export default AutoCompleteListItem;
