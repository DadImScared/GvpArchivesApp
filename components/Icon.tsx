import * as React from "react";

import { Icon as NativeBaseIcon, NativeBase } from "native-base";

export class Icon extends React.Component<NativeBase.Icon> {
  render() {
    return (
      <NativeBaseIcon {...this.props} />
    );
  }
}

export default Icon;
