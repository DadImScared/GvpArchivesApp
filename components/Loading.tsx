import * as React from "react";

import { Spinner } from "native-base";
import { View } from "react-native";

export const Loading: React.FunctionComponent<{ isLoading: boolean }> = ({ isLoading }) => isLoading ? (
  <View>
    <Spinner />
  </View>
) : null;

export default Loading;
