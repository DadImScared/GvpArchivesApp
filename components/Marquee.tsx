import * as React from "react";

import Ticker, { ITextMarqueeProps } from "react-native-text-ticker";

const ForwardedMarquee = React.forwardRef<Ticker, ITextMarqueeProps>(({ children, ...other }, ref) => (
  <Ticker {...other} ref={ref}>
    {children}
  </Ticker>
));

export default ForwardedMarquee;
