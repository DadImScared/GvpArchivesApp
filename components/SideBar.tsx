import * as React from "react";

import { NavigationScreenProps, StackActions } from "react-navigation";

import { Container, Content } from "native-base";

import MenuItem from "./MenuItem";

export interface IRouteComponent {
  component: any;
  displayName: string;
}

export type IRoute = string | IRouteComponent;

export const routes: { [key: string]: IRoute } = {
  bhagavatpatrika: "Bhagavat Patrika",
  book: "Books",
  harikatha: "Harikatha",
  harmonistmagazine: "Harmonist Magazine",
  harmonistmonthly: "Harmonist Monthly",
  lecture: "Lectures",
  movie: "Movies",
  song: "Songs"
};

const routesArray = Object.entries(routes);

export const getActiveRoute = (navigationState: any): any => {
  if (!navigationState) {
    return null;
  }
  const route = navigationState.routes[navigationState.index];
  // dive into nested navigators
  if (route.routes) {
    return getActiveRoute(route);
  }
  return route;
};

export class SideBar extends React.Component<NavigationScreenProps, { activeRoute: string }> {

  constructor(props: NavigationScreenProps) {
    super(props);
    this.state = {
      activeRoute: "Home"
    };
  }

  navigateRoute = (to: string) => {
    const { navigation } = this.props;
    const { activeRoute } = this.state;
    navigation.closeDrawer();
    if (to === activeRoute || !to) return;

    navigation.dispatch(StackActions.push({
      params: {
        category: to
      },
      routeName: "ItemsByCategory"
    }));
  };

  componentDidUpdate(
    prevProps: Readonly<NavigationScreenProps>,
    prevState: Readonly<{ activeRoute: string }>,
    snapshot?: any
  ): void {
    const route = getActiveRoute(this.props.navigation.state);
    if (route.routeName === "ItemsByCategory") {
      if (this.state.activeRoute !== (route.params as any).category) {
        this.setState({ activeRoute: (route.params as any).category });
      }
    }
    else if (this.state.activeRoute) {
      this.setState({ activeRoute: "" });
    }
  }

  shouldComponentUpdate(
    nextProps: Readonly<NavigationScreenProps>,
    nextState: Readonly<{ activeRoute: string }>, nextContext: any
  ): boolean {
    const { navigation: { state: { index: nextIndex, routes: nextRoutes } } } = nextProps;
    const { navigation: { state: { index: currentIndex, routes: currentRoutes } } } = this.props;
    const currentRoute = currentRoutes[currentIndex].routes[currentRoutes[currentIndex].index].key;
    const nextRoute = nextRoutes[nextIndex].routes[nextRoutes[nextIndex].index].key;
    return nextRoute !== currentRoute || (this.state.activeRoute !== nextState.activeRoute);
  }

  renderRow = ([to, displayName]: [string, IRoute]) => {
    return (
      <MenuItem
        key={to}
        active={to === this.state.activeRoute}
        navigateRoute={this.navigateRoute}
        to={to}
        display={displayName}
      />
    );
  };

  render() {
    return (
        <Container>
          <Content>
            {routesArray.map((route) => this.renderRow(route))}
          </Content>
        </Container>
    );
  }
}

export default SideBar;
