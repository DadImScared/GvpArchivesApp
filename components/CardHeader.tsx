import * as React from "react";

import { Body, CardItem, Text } from "native-base";
import { routes } from "./SideBar";

interface IProps {
  item: any;
  extraSubHeader?: string;
}

export const CardHeader: React.FunctionComponent<IProps> = ({ item, extraSubHeader }) => (
  <CardItem header={true}>
    <Body>
      <Text>{item.title.trim()}</Text>
      <Text note={true}>{routes[item.category]}{extraSubHeader ? `${extraSubHeader}` : null}</Text>
    </Body>
  </CardItem>
);

export default CardHeader;
