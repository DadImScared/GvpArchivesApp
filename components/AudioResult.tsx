import * as React from "react";

import { Button, Card, CardItem, Left } from "native-base";
import { StyleSheet, Text } from "react-native";

import { Item } from "../reducers/itemsById";
import CardHeader from "./CardHeader";

export interface IAudioResultProps {
  item: Item;
  [key: string]: any;
}

const styles: any = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center",
    margin: 4,
    width: "30%"
  }
});

export const AudioResult: React.FC<IAudioResultProps> = ({ item }) => (
  <Card>
    <CardHeader item={item} />
    <CardItem footer={true}>
      <Left>
        <Button primary={true} style={styles.button}><Text style={{ color: "white" }}>Play</Text></Button>
        <Button info={true} style={styles.button}><Text>Playlist</Text></Button>
      </Left>
    </CardItem>
  </Card>
);

export default AudioResult;
