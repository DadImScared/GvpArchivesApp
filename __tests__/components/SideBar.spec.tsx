import * as React from "react";
import * as renderer from "react-test-renderer";

import { NavigationScreenProps, StackActions } from "react-navigation";

import { SideBar } from "../../components/SideBar";

let props: NavigationScreenProps;

describe("SideBar", () => {
  beforeEach(() => {
    props = {
      navigation: {
        closeDrawer: jest.fn(),
        dispatch: jest.fn()
      } as any
    };
  });

  it("should render", () => {
    const tree = renderer.create(<SideBar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should navigateRoute", () => {
    const tree = renderer.create(<SideBar {...props} />);
    const instance = tree.root;
    instance.instance.navigateRoute("book");
    expect(props.navigation.dispatch).toHaveBeenCalledWith(StackActions.push({
      params: {
        category: "book"
      },
      routeName: "ItemsByCategory"
    }));
    expect(props.navigation.closeDrawer).toBeCalled();
  });
});
