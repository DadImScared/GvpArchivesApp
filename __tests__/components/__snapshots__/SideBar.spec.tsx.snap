// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
/>
`;
